Foundation/JSONEncoder.swift

```diff
@@ -270,7 +272,7 @@ fileprivate struct _JSONEncodingStorage {
     }
 
     fileprivate mutating func popContainer() -> NSObject {
-        precondition(!self.containers.isEmpty, "Empty container stack.")
+        precondition(self.containers.count > 0, "Empty container stack.")
```

Comment:
@saiHemak this is also wrong: https://github.com/apple/swift/blob/master/stdlib/public/SDK/Foundation/JSONEncoder.swift#L359